oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/DefaultOAuth2AuthenticatedPrincipal.java

```diff
@@ -64,7 +64,7 @@ public DefaultOAuth2AuthenticatedPrincipal(String name, Map<String, Object> attr
 		this.attributes = Collections.unmodifiableMap(attributes);
 		this.authorities = (authorities != null) ? Collections.unmodifiableCollection(authorities)
 				: AuthorityUtils.NO_AUTHORITIES;
-		this.name = (name != null) ? name : (String) this.attributes.get("sub");
+		this.name = (name != null) ? name : OAuth2AuthenticatedPrincipal.super.getName();
```

Comment:
Let's please leave this as-is.

Reply:
I've reverted the line to

```java
this.name = (name != null) ? name : (String) this.attributes.get("sub");
```

`"sub"` is now used twice within `DefaultOAuth2AuthenticatedPrincipal`.